network/network.go

```diff
@@ -0,0 +1,261 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Network Stats provider
+package network
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/intelsdi-x/snap-plugin-collector-docker/mounts"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/wrapper"
+	"github.com/intelsdi-x/snap-plugin-utilities/ns"
+)
+
+const networkInterfacesDir = "/sys/class/net"
+
+var networkMetrics = getListOfNetworkMetrics()
+
+func getListOfNetworkMetrics() []string {
+	metrics := []string{}
+	ns.FromCompositionTags(wrapper.NetworkInterface{}, "", &metrics)
+	return metrics
+}
+
+func NetworkStatsFromProc(rootFs string, pid int) (ifaceStats []wrapper.NetworkInterface, errout error) {
+	netStatsFile := filepath.Join(rootFs, mounts.ProcfsMountPoint, strconv.Itoa(pid), "/net/dev")
+	var err error
+	ifaceStats, err = scanInterfaceStats(netStatsFile)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't read network stats: %v", err)
+	}
+
+	if len(ifaceStats) == 0 {
+		return nil, errors.New("No network interface found")
+	}
+
+	// calculate network stats in total (range over all net interfaces)
+	total := wrapper.NetworkInterface{
+		Name: "total",
+	}
+	for _, iface := range ifaceStats {
+		total.RxBytes += iface.RxBytes
+		total.RxPackets += iface.RxPackets
+		total.RxDropped += iface.RxDropped
+		total.RxErrors += iface.RxErrors
+		total.TxBytes += iface.TxBytes
+		total.TxPackets += iface.TxPackets
+		total.TxDropped += iface.TxDropped
+		total.TxErrors += iface.TxErrors
+	}
+	ifaceStats = append(ifaceStats, total)
+
+	return ifaceStats, nil
+}
+
+func NetworkStatsFromRoot() (ifaceStats []wrapper.NetworkInterface, _ error) {
+	devNames, err := listRootNetworkDevices()
+	if err != nil {
+		return nil, err
+	}
+	ifaceStats = []wrapper.NetworkInterface{}
+	for _, name := range devNames {
+		if isIgnoredDevice(name) {
+			continue
+		}
+		if stats, err := interfaceStatsFromDir(name); err != nil {
+			return nil, err
+		} else {
```

Comment:
`else` block seems to be unnecessary because `if` block ends with return statement